vivisect/analysis/elf/elfplt.py

```diff
@@ -22,16 +66,99 @@ def analyze(vw):
             continue
         analyzePLT(vw, sva, ssize)
 
+gots = {}
+def getGOT(vw, fileva):
+    '''
+    Returns GOT location for the file which contains the address fileva.
+
+    First checks through Sections (vw.getSegments), then through Dynamics.
+    If the two clash, Dynamics wins.
+
+    Linux binaries tend to have both Sections and Dynamics entries
+    OpenBSD is only Sections.
+    QNX has only Dynamics.
+    '''
+    global gots
+    filename = vw.getFileByVa(fileva)
+
+    gottup = gots.get(filename)
+    if gottup is not None:
+        return gottup
+
+    for va, size, name, fname in vw.getSegments():
+        if name == ".got" and fname == filename:
+            gotva = va
+            gotsize = size
+            break
+
+    # pull GOT info from Dynamics
+    fdyns = vw.getFileMeta(filename, 'ELF_DYNAMICS')
+    if fdyns is not None:
+        FGOT = fdyns.get('DT_PLTGOT')
+        if vw.getFileMeta(filename, 'addbase'):
+            imgbase = vw.getFileMeta(filename, 'imagebase')
+            logger.debug('Adding Imagebase: 0x%x', imgbase)
+            FGOT += imgbase
+        if gotva is not None:
+            if gotva != FGOT:
+                logger.warn("Dynamics and Sections have different GOT entries: S:0x%x D:0x%x. using Dynamics", gotva, FGOT)
+                # since Dynamics don't store the GOT size, just use to the end of the memory map
+                mmva, mmsz, mmperm, mmname = vw.getMemoryMap(FGOT)
+                moffset = gotva - mmva
+                gotsize = mmsz - moffset
+            gotva = FGOT
+
+    gots[filename] = (gotva, gotsize)
+    return gotva, gotsize
+
+def getPLTs(vw):
```

Comment:
Also, who uses this?

Reply:
nobody yet.  it's going in a couple places, but i needed a consistent place to get a list of PLT sections for a given workspace/file, because QNX doesn't *do* ELF Sections... only Dynamics.

Reply:
incorporated now.